assets/developer-sync.html

```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Developer Sync</title>
+  </head>
+  <body>
+    <script type="text/javascript">
+      // Listen for postMessage events from Braze domains
+    window.addEventListener('message', function(event) {
+        const isAllowedOrigin = /^https:\/\/.*\.braze\.(com|eu)$/.test(event.origin) || /^http:\/\/localhost:/.test(event.origin);
+        // Check if the origin is from a braze.com or braze.eu domain
+        if (isAllowedOrigin) {
+          // Process the message
+          if (event.data && event.data?.type === 'BRAZE_DEVELOPER_DATA') {
+              window.localStorage.setItem('braze_developer_data', JSON.stringify(event.data.payload));
+          }
+        }
+    });
```

Comment:
this is probably better done without using test
(double check this works)

```suggestion
   window.addEventListener('message', (event) => {
  try {
    const url = new URL(event.origin);
    const { protocol, hostname } = url;
  
    // Check if it's from localhost (http only)
    const isLocalhost = protocol === 'http:' && hostname.startsWith('localhost');
  
    // Check if it's from a Braze domain (must be https)
    const isBrazeDomain =
      protocol === 'https:' &&
      (hostname.endsWith('.braze.com') || hostname.endsWith('.braze.eu'));
  
    if (!isLocalhost && !isBrazeDomain) return;
  
    // Process the message
    const { type, payload } = event.data || {};
    if (type === 'BRAZE_DEVELOPER_DATA' && payload) {
      window.localStorage.setItem('braze_developer_data', JSON.stringify(payload));
    }
  } catch (e) {
    // event.origin might not be a valid URL
    console.error('Invalid origin:', event.origin);
  }
});
```